src/app/ReadClient.cpp

```diff
@@ -493,6 +498,18 @@ void ReadClient::OnActiveModeNotification()
         return;
     }
 
+    // When receiving check-in message, it means all tracked subscriptions for this node has gone in server side, if there is a related active subscription
+    // and subscription has not yet rescheduled in client side, we should forcibly timeout the current subscription, and schedule a new one.
+    if (!mIsResubscriptionScheduled)
+    {
+        // Closing will ultimately trigger ScheduleResubscription with the aReestablishCASE argument set to true, effectively
+        // rendering the session defunct.
+        Close(CHIP_ERROR_TIMEOUT);
+        return;
+    }
+
+    // If the server sends a check-in message and a subscription is already scheduled, it indicates a client-side subscription
+    // timeout or failure. Cancel the scheduled subscription and initiate a new one immediately.
```

Comment:
```suggestion
    // If we have already detected subscription loss and are waiting to try to re-subscribe,
    // now is a really good time to do it, since the server is listening.
```